### PR TITLE
Deeplink to send payments, token amount better accuracy, update OkHttp

### PR DESCRIPTION
**Pull request by MrStahlfelge**

### CODE REVIEW

File: app/src/main/java/org/ergoplatform/android/transactions/ChooseSpendingWalletFragmentDialog.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+        AppDatabase.getInstance(requireContext()).walletDao().getWalletsWithStates()
+            .observe(viewLifecycleOwner, {
+                binding.listWallets.removeAllViews()
+
+                val walletsWithoutReadonly = it.filter { it.walletConfig.secretStorage != null }
+
+                if (walletsWithoutReadonly.size == 1) {
+                    // immediately switch to send funds screen
+                    navigateToSendFundsScreen(it.first().walletConfig.id, true)
```
Comment (aslesarenko) [inline app/src/main/java/org/ergoplatform/android/transactions/ChooseSpendingWalletFragmentDialog.kt:62]:
```suggestion
                    navigateToSendFundsScreen(walletsWithoutReadonly.first().walletConfig.id, true)
```

File: app/src/main/java/org/ergoplatform/android/ui/UiUtils.kt [hunk 2]
```diff
@@ -34,6 +34,14 @@ fun inputTextToFloat(amountStr: String?): Float {
     }
 }
 
+fun inputTextToDouble(amountStr: String?): Double {
+    try {
+        return if (amountStr == null || amountStr.isEmpty()) 0.0 else amountStr.toDouble()
```
Comment (aslesarenko) [inline app/src/main/java/org/ergoplatform/android/ui/UiUtils.kt:39]: I remember there is `isNullOrEmpty` method.

**Review by aslesarenko — APPROVED** [review]
LGTM, but please see comments

File: app/src/main/java/org/ergoplatform/android/transactions/ChooseSpendingWalletFragmentDialog.kt (hunk 1 above)
Comment (aslesarenko) [reply app/src/main/java/org/ergoplatform/android/transactions/ChooseSpendingWalletFragmentDialog.kt:62]: looks like a bug

Comment (MrStahlfelge) [reply app/src/main/java/org/ergoplatform/android/transactions/ChooseSpendingWalletFragmentDialog.kt:62]: Thank you, very good catch

**Pull request merged**